packaging/windows/lbry-win32-app/LBRYWin32App.py

```diff
@@ -0,0 +1,283 @@
+                self.create_menu(submenu, option_action)
+                item, extras = win32gui_struct.PackMENUITEMINFO(text=option_text,
+                                                                hbmpItem=option_icon,
+                                                                hSubMenu=submenu)
+                win32gui.InsertMenuItem(menu, 0, 1, item)
+
+    def prep_menu_icon(self, icon):
+        # First load the icon.
+        ico_x = win32api.GetSystemMetrics(win32con.SM_CXSMICON)
+        ico_y = win32api.GetSystemMetrics(win32con.SM_CYSMICON)
+        hicon = win32gui.LoadImage(0, icon, win32con.IMAGE_ICON, ico_x, ico_y, win32con.LR_LOADFROMFILE)
+
+        hdcBitmap = win32gui.CreateCompatibleDC(0)
+        hdcScreen = win32gui.GetDC(0)
+        hbm = win32gui.CreateCompatibleBitmap(hdcScreen, ico_x, ico_y)
+        hbmOld = win32gui.SelectObject(hdcBitmap, hbm)
+        # Fill the background.
+        brush = win32gui.GetSysColorBrush(win32con.COLOR_MENU)
+        win32gui.FillRect(hdcBitmap, (0, 0, 16, 16), brush)
+        # unclear if brush needs to be feed.  Best clue I can find is:
+        # "GetSysColorBrush returns a cached brush instead of allocating a new
+        # one." - implies no DeleteObject
+        # draw the icon
+        win32gui.DrawIconEx(hdcBitmap, 0, 0, hicon, ico_x, ico_y, 0, 0, win32con.DI_NORMAL)
+        win32gui.SelectObject(hdcBitmap, hbmOld)
+        win32gui.DeleteDC(hdcBitmap)
+
+        return hbm
+
+    def command(self, hwnd, msg, wparam, lparam):
+        id = win32gui.LOWORD(wparam)
+        self.execute_menu_option(id)
+
+    def execute_menu_option(self, id):
+        menu_action = self.menu_actions_by_id[id]
+        if menu_action == self.QUIT:
+            win32gui.DestroyWindow(self.hwnd)
+        else:
+            menu_action(self)
+
+
+def main():
+    def LBRYApp():
+        SysTrayIcon(icon, hover_text, menu_options, on_quit=stop)
+
+    def openui_(sender):
+        webbrowser.open(UI_ADDRESS)
+
+    def replyToApplicationShouldTerminate_():
+        reactor.stop()
+
+    def stop(sysTrayIcon):
+        replyToApplicationShouldTerminate_()
+
+    if getattr(sys, 'frozen', False) and os.name == "nt":
+        icon = os.path.join(os.path.dirname(sys.executable), ICON_PATH, 'lbry16.ico')
+    else:
+        icon = os.path.join(ICON_PATH, 'lbry16.ico')
+    hover_text = APP_NAME
+    menu_options = (('Open', icon, openui_),)
+
+    if not test_internet_connection():
+        log.warn('No Internet Connection')
+        sys.exit(1)
+
+    systray_thread = threading.Thread(target=LBRYApp)
+    systray_thread.start()
+
+    lbry = LBRYDaemonServer()
+    d = lbry.start()
+    d.addCallback(lambda _: webbrowser.open(UI_ADDRESS))
+    lbrynet_server = server.Site(lbry.root)
+    lbrynet_server.requestFactory = LBRYDaemonRequest
+    reactor.listenTCP(API_PORT, lbrynet_server, interface=API_INTERFACE)
+    # reactor.addSystemEventTrigger("after", "shutdown", AppHelper.stopEventLoop)
```

Comment:
The program ends cleanly on api.stop() call now that the systray icon runs as a daemon thread